### PR TITLE
Add argument of `alpha` to `MixUpSoftLabelDataset`

### LINKED ISSUE #821: Add `alpha` argument to `MixUpSoftLabelDataset`
In the original paper of mixup (https://arxiv.org/pdf/1710.09412.pdf), λ (`mix_ratio` in the code) is sampled from Beta(α, α), where α is a hyperparameter. On the other hand, in ChainerCV's `MixUpSoftLabelDataset`, `mix_ratio` is sampled from uniform distribution (which is the same as α=1 case). It might be more useful if we could specify the value of α. 

https://github.com/chainer/chainercv/blob/6ce2d47f4b1992a0e257392f75312547b268eb5f/chainercv/datasets/mixup_soft_label_dataset.py#L67

### PR DESCRIPTION
**Pull request by crcrpar**
Resolves #821.

### CODE REVIEW

**Comment by yuyu2172** [comment]
Thank you so much.
Could you add test that uses `alpha` option?

File: chainercv/datasets/mixup_soft_label_dataset.py [hunk 1]
```diff
@@ -55,16 +60,21 @@ class MixUpSoftLabelDataset(dataset_mixin.DatasetMixin):
     .. [#mixup_1] Same as :obj:`dataset`.
     """
 
-    def __init__(self, dataset, n_class):
+    def __init__(self, dataset, n_class, alpha=1.0):
+        if alpha <= 0.0:
```
Comment (crcrpar) [inline chainercv/datasets/mixup_soft_label_dataset.py]: note: NumPy documentation says `alpha` is non-negative but it rejects alpha of 0 raising Value Error of `a <= 0`.

**Comment by yuyu2172** [comment]
LGTM!

**Comment by yuyu2172** [comment]
pfnCI, test this please

**Pull request merged**